test/extended/networking/services.go

```diff
@@ -65,4 +69,63 @@ var _ = Describe("[sig-network] services", func() {
 			Expect(checkServiceConnectivity(f1, f2, DIFFERENT_NODE)).To(Succeed())
 		})
 	})
+
+	InBareMetalClusterContext(oc, func() {
+		f := e2e.NewDefaultFramework("external-ip-auto-assign")
+
+		It("ensures ingress ip not assigned for service when external ip config not set [Serial]", func() {
+			By("update network config without external ip")
+			modifyNetworkConfig(oc, []string{}, []string{}, []string{})
+			By("create service of type load balancer")
+			serviceName := "test-service-without-external-ip"
+			launchForLoadbalancerTypeService(f, serviceName, "", []string{})
+			By("check load balance service not having ingress ip configured")
+			Eventually(func() string {
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
+				expectNoError(err)
+				if len(service.Status.LoadBalancer.Ingress) == 0 {
+					return ""
+				}
+				return service.Status.LoadBalancer.Ingress[0].IP
+			}, 5*time.Minute, 1*time.Second).Should(Equal(""))
+		})
+
+		It("ensures service external ip is accepted when allowedCIDRs is set [Serial]", func() {
+			By("update network config with allowedCIDRs for external ip")
+			modifyNetworkConfig(oc, nil, []string{"192.168.132.10/32"}, nil)
+			By("create service of type load balancer")
+			serviceName := "test-service-with-external-ip"
+			launchForLoadbalancerTypeService(f, serviceName, "", []string{"192.168.132.10"})
+			By("check load balance service having desired ingress ip configured")
+			Eventually(func() string {
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
```

Comment:
The ExternalIPRanger doesn't work this way. Rather, it is an admission controller that only enforces on Create / Update. Also, it takes 5-10 minutes for the changes to roll out.

So, you need to re-write this test to work like this:
1 check that creating a service of type ClusterIP with an ExternalIP set doesn't work
2. add a cidr to the allow config
3. try creating a service over and over until it does work
4. try creating a service outside the range, see that it doesn't work
5. remove the cidr from the allow config
6. try creating a service w/ ExternalIP over and over until it doesn't work

Reply:
ok @squeed as discussed, i think test is changed as per the suggestion above, let me see how test runs now.